### PR TITLE
When querying using unsupported resource value throw

### PR DESCRIPTION
**Pull request by aelred**

### CODE REVIEW

**Comment by aelred** [comment]
please retest this please

**Comment by aelred** [comment]
plesae retest this please - another timeout: http://10.0.1.19:9998/job/unit-test-pull-request-grakn-titan/3506/

**Comment by aelred** [comment]
please retest this please - another random failure: http://10.0.1.19:9998/job/unit-test-pull-request-grakn-titan/3528/ai.grakn$grakn-test/testReport/ai.grakn.test.engine.controller/GraqlControllerTest/POSTGraqlInsert_InsertWasExecutedOnGraph/

**Comment by aelred** [comment]
please retest this please - another random failure: http://10.0.1.19:9998/job/unit-test-pull-request-grakn-titan/ai.grakn$grakn-test/3534/testReport/junit/ai.grakn.test.engine.controller/GraqlControllerTest/GETGraqlAggregateWithTextType_ResponseIsCorrect/

**Comment by aelred** [comment]
please retest this please - analytics is sad: http://10.0.1.19:9998/job/unit-test-pull-request-grakn-titan/ai.grakn$grakn-test/3543/testReport/junit/ai.grakn.test.graql.analytics/ClusteringTest/testConnectedComponentConcurrency/

File: grakn-graql/src/test/java/ai/grakn/graql/internal/query/predicate/ComparatorPredicateTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    public void whenPassedALocalDateTime_DontThrow() {
+        new MyComparatorPredicate(LocalDate.of(2012, 12, 21).atStartOfDay());
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void whenPassedANull_Throw() {
+        new MyComparatorPredicate(null);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
```
Comment (fppt) [inline grakn-graql/src/test/java/ai/grakn/graql/internal/query/predicate/ComparatorPredicateTest.java]: Nitpick police. Can you check the contents of the actual error message as well. It makes things easier for me when I break graql and I have to debug. It gives me a more clearly defined starting point.

File: grakn-test/src/test/java/ai/grakn/test/graql/query/QueryErrorTest.java [hunk 2]
```diff
@@ -182,4 +182,9 @@ public void testGetNonExistentVariable() {
         //noinspection ResultOfMethodCallIgnored
         concepts.count();
     }
+
+    @Test(expected = IllegalArgumentException.class)
```
Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/graql/query/QueryErrorTest.java]: Nitpick again.

**Comment by aelred** [comment]
!rtg http://10.0.1.19:9998/job/unit-test-pull-request-grakn-titan/ai.grakn$grakn-test/3608/testReport/junit/ai.grakn.test.engine.postprocessing/PostProcessingIT/checkThatDuplicateResourcesAtLargerScaleAreMerged/

**Comment by aelred** [comment]
!rtg

**Comment by aelred** [comment]
@fppt please re-review my beautiful new tests

**Pull request merged**